src/test/njs_unit_test.c

```diff
@@ -16882,6 +16912,115 @@ static njs_unit_test_t  njs_test[] =
     { njs_str("Date.constructor === Function"),
       njs_str("true") },
 
+    { njs_str("var y = Object.defineProperty({}, Symbol.toPrimitive, {"
+              "    get: function() {"
+              "        throw new Error('expected');"
+              "    }"
+              "});"
+              "new Date(y);"),
+      njs_str("Error: expected") },
+
+    { njs_str("var y = {};"
+              "y[Symbol.toPrimitive] = function() {"
+              "     throw new Error('expected');"
+              "};"
+              "new Date(y);"),
+      njs_str("Error: expected") },
+
+    { njs_str("var y = {};"
+              "var callCount = 0;"
+              "var thisVal, args;"
+              "y[Symbol.toPrimitive] = function() {"
+              "    callCount += 1;"
+              "    thisVal = this;"
+              "    args = arguments;"
+              "};"
+              "new Date(y);"
+              "callCount === 1 && thisVal === y && args.length === 1 && args[0] === 'default'"),
+      njs_str("true") },
+
+    { njs_str("var y = {};"
+              "var retVal;"
+              "y[Symbol.toPrimitive] = function() {"
+              "    return retVal;"
+              "};"
+              "retVal = {};"
+              "new Date(y);"),
+      njs_str("TypeError") },
+
+    { njs_str("var badToPrimitive = {};"
+              "badToPrimitive[Symbol.toPrimitive] = function() {"
+              "    throw new Error('expected');"
+              "};"
+              "new Date(badToPrimitive);"),
+      njs_str("Error: expected") },
+
+    { njs_str("var spyToPrimitive = {};"
+              "var callCount = 0;"
+              "var thisValue, args;"
+              "spyToPrimitive[Symbol.toPrimitive] = function() {"
+              "    thisValue = this;"
+              "    args = arguments;"
+              "    callCount += 1;"
+              "};"
+              "new Date(spyToPrimitive);"
+              "callCount === 1 && thisValue === spyToPrimitive && args.length === 1 && args[0] === 'default';"),
+      njs_str("true") },
+
+    { njs_str("var poisonedObject = {};"
+              "var poisonedDate = new Date();"
+              "Object.defineProperty(poisonedObject, Symbol.toPrimitive, {"
+              "    get: function() {"
+              "        throw new Error('expected');"
+              "    }"
+              "});"
+              "Object.defineProperty(poisonedDate, Symbol.toPrimitive, {"
+              "    get: function() {"
+              "        throw new Test262Error();"
```

Comment:
Test262Error() is not defined here. 
It is better to replace it with Error("poisonedDate topPrimitive") or something.
Otherwise if poisonedDate.topPrimitive is called ReferenceError will be printed which can be confusing.